### ISSUE TITLE
Support for removing certain attribute from JSX Element using babel-plugin-transform-react-jsx

### ISSUE THREAD

**Issue by asingh04** [babel/babel#13141]
## Feature Request

- [x] I would like to work on this feature!

**Is your feature request related to a problem?**
It is common to use custom attributes to help in selecting some elements by automated testing tools like Selenium or Cyprus or by unit testing tools. However, these custom attributes, appearing in the DOM in the production environment, are redundant.

**Describe the solution you'd like**
We can specify the attribute to remove or strip as an option to the `@babel/plugin-transform-react-jsx` in the production configuration. This can help in retaining those custom attributes only in the development/staging environment where automation or unit tests can run.

**Describe alternatives you've considered**.

**Documentation, Adoption, Migration Strategy**
```js
{
  "plugins": [
    [
      "@babel/plugin-transform-react-jsx",
      {
        "throwIfNamespace": false, // defaults to true
        "runtime": "automatic", // defaults to classic
        "importSource": "custom-jsx-library", // defaults to react
        "removeAttribute": "data-auto-id" // <-- new option to the plugin which can be a simple string or a regular experssion
      }
    ]
  ]
}
```

**Comment by nicolo-ribaudo** [comment]
You can try doing it with a custom plugin, that you only enable in production.

**Comment by asingh04** [comment]
I initially thought of making a custom plugin to do so, but I think this is a very common problem for most of the developers using react and this functionality of removing some attributes from the final dom could be provided from the babel transform plugin.

**Comment by proteriax** [comment]
I agree that a custom plugin is the way to go, because you can trivially just match against JSXAttribute. Removing an attribute seems out of scope and unnecessarily complicate the code.
